dataplane/ebpf/src/egress/tcp.rs

```diff
@@ -68,6 +71,39 @@ pub fn handle_tcp_egress(ctx: TcContext) -> Result<i32, i64> {
     unsafe { (*tcp_hdr).check = 0 };
 
     // TODO: connection tracking cleanup https://github.com/kubernetes-sigs/blixt/issues/85
+    // SNAT the port
+    unsafe { (*tcp_hdr).source = tcp_backend.backend_key.port as u16 };
+
+    let tcp_hdr_ref = unsafe { tcp_hdr.as_ref().ok_or(TC_ACT_OK)? };
+
+    // If the packet has the RST flag set, it means the connection is being terminated, so remove it
+    // from our map.
+    if tcp_hdr_ref.rst() == 1 {
+        unsafe {
+            TCP_CONNECTIONS.remove(&client_key)?;
+        }
+    }
+
+    let mut tcp_state = tcp_backend.state;
+    let moved = handle_tcp_conn_close(tcp_hdr_ref, &mut tcp_state);
+    // If the connection has moved to the Closed state, stop tracking it.
+    if let TCPState::Closed = tcp_state {
+        unsafe {
+            TCP_CONNECTIONS.remove(&client_key)?;
+        }
+    // If the connection has not reached the Closed state yet, but it did advance to a new state,
+    // then record the new state.
+    } else if moved {
```

Comment:
so from what i can understand, we modify `handle_tcp_conn_close` to accept an immutable reference to the backend's state and then return a new state if needed?
this is what i had initially, but decided to change it to the current logic mainly to avoid having to create a new object and reduce memory usage. but then again, this is just an enum so it might be a insignificant optimization. wdyt?

Reply:
> so from what i can understand, we modify handle_tcp_conn_close to accept an immutable reference to the backend's state and then return a new state if needed?

Yeah, this is kinda what I was leaning towards? But I wasn't feeling super strongly about it, I was more curious what your thoughts were.

>  this is what i had initially, but decided to change it to the current logic mainly to avoid having to create a new object and reduce memory usage. but then again, this is just an enum so it might be a insignificant optimization. wdyt?

I kinda started down the path of doing a performance analysis here for us to consider together, but quickly realized it would be time consuming and that it shouldn't hold us up from getting something that _works_ first. So let's not worry to much about optimizations at this level just yet, I'm satisfied with what's already here and we can come back around and look these things over in later iterations. :+1: